src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -662,6 +662,50 @@ def info
       {'stemcell_formats' =>  ['vsphere-ovf', 'vsphere-ova']}
     end
 
+    #creates T1 router and virtual switch attached to it
+    def create_subnet(subnet_definition)
+      raise 'NSXT must be enabled in CPI to use create_subnet' if !@config.nsxt_enabled?
+      cloud_properties = subnet_definition['cloud_properties']
+      raise 'cloud_properties must be provided' if cloud_properties.nil?
```

Comment:
Input validation should be moved into its own class. We're moving away from the use of passing schemed objects as hashes. See `VmConfig` or `VmType`.